File: Archivist/src/test/java/com/zorroa/archivist/service/ImageServiceTests.java
package com.zorroa.archivist.service;

import com.zorroa.archivist.ArchivistApplicationTests;
import com.zorroa.archivist.sdk.service.ImageService;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.io.File;
import java.io.IOException;

public class ImageServiceTests extends ArchivistApplicationTests {

    @Autowired
    ImageService imageService;

    @Test
    public void testGenerateProxyPath() throws IOException {
        String id = "ABCD1234-ABCD1234-ABCD1234-ABCD1234";
        File path = imageService.allocateProxyPath(id, "png");
        logger.info(path.getAbsolutePath());
    }
}

File: services/archivist/archivist/src/test/kotlin/com/zorroa/archivist/service/ImageServiceTests.kt
package com.zorroa.archivist.service

import com.zorroa.archivist.AbstractTest
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import java.io.File
import kotlin.test.assertEquals

class ImageServiceTests : AbstractTest() {

    @Autowired
    lateinit var imageService: ImageService

    @Test
    fun testGetImageDimension() {
        val dim = imageService.getImageDimension(File("src/test/resources/test-data/toucan.jpg"))
        assertEquals(512, dim.width)
        assertEquals(341, dim.height)
    }
}